load-testing/tests/relays_stress_helpers_test.go

```diff
@@ -0,0 +1,1143 @@
+	fundingAccountKeyRecord, err := s.txContext.GetKeyring().Key(fundingAccountKeyName)
+	require.NoError(s, err)
+
+	fundingAccountAddress, err := fundingAccountKeyRecord.GetAddress()
+	require.NoError(s, err)
+
+	s.fundingAccountInfo = &accountInfo{
+		keyName:     fundingAccountKeyName,
+		accAddress:  fundingAccountAddress,
+		pendingMsgs: []sdk.Msg{},
+	}
+}
+
+// initializeProvisionedActors parses the load test manifest and initializes the
+// gateway and supplier keyNames and the URLs used to send requests to.
+func (s *relaysSuite) initializeProvisionedActors() {
+	loadTestManifestContent, err := os.ReadFile(loadTestManifestPath)
+	require.NoError(s, err)
+
+	provisionedActors, err := config.ParseLoadTestManifest(loadTestManifestContent)
+	require.NoError(s, err)
+
+	for _, gateway := range provisionedActors.Gateways {
+		s.gatewayUrls[gateway.KeyName] = gateway.ExposedUrl
+	}
+
+	for _, supplier := range provisionedActors.Suppliers {
+		s.suppliersUrls[supplier.KeyName] = supplier.ExposedUrl
+	}
+}
+
+// TODO_IN_THIS_COMMIT: godoc comment.
+func (s *relaysSuite) mapSessionInfoFn(
+	batchInfoPublishCh chan<- *batchInfoNotif,
+) channel.MapFn[client.Block, *sessionInfoNotif] {
+	var (
+		// The test suite is initially waiting for the next session to start.
+		waitingForFirstSession = true
+		prevBatchTime          time.Time
+	)
+
+	return func(
+		ctx context.Context,
+		block client.Block,
+	) (*sessionInfoNotif, bool) {
+		blockHeight := block.Height()
+		if blockHeight <= s.latestBlock.Height() {
+			return nil, true
+		}
+
+		sessionInfo := &sessionInfoNotif{
+			blockHeight:             blockHeight,
+			sessionNumber:           keeper.GetSessionNumber(blockHeight),
+			sessionStartBlockHeight: keeper.GetSessionStartBlockHeight(blockHeight),
+			sessionEndBlockHeight:   keeper.GetSessionEndBlockHeight(blockHeight),
+		}
+
+		infoLogger := logger.Info().
+			Int64("session_num", sessionInfo.sessionNumber).
+			Int64("block_height", block.Height())
+
+		// If the test has not started and the current block is not the first block
+		// of the session, wait for the next session to start.
+		if waitingForFirstSession && blockHeight != sessionInfo.sessionStartBlockHeight {
+			countDownToTestStart := sessionInfo.sessionEndBlockHeight - blockHeight + 1
+			infoLogger.Msgf(
+				"waiting for next session to start: in %d blocks",
+				countDownToTestStart,
+			)
+
+			// The test is not to be started yet, skip the notification to the downstream
+			// observables until the first block of the next session is reached.
+			return nil, true
+		}
+
+		// If the test has not started, set the start block height to the current block height.
+		// As soon as the test start, s.startBlockHeight will no longer be updated.
+		// It is updated only once at the start of the test.
+		if waitingForFirstSession {
+			s.startBlockHeight = blockHeight
+		}
+
+		// Mark the test as started.
+		waitingForFirstSession = false
```

Comment:
It doesn't make a difference practically, but logically I feel like this should be inside the if statement above.